### PR TITLE
Move default unmarshaler to service to avoid using deprecate configs

### PR DESCRIPTION
**Pull request by bogdandrutu**
Remove already deprecates `config.Service`.

### CODE REVIEW

File: service/unmarshaler.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+type configSettings struct {
+	Receivers  *configunmarshaler.Receivers  `mapstructure:"receivers"`
+	Processors *configunmarshaler.Processors `mapstructure:"processors"`
+	Exporters  *configunmarshaler.Exporters  `mapstructure:"exporters"`
+	Extensions *configunmarshaler.Extensions `mapstructure:"extensions"`
+	Service    ConfigService                 `mapstructure:"service"`
+}
+
+// unmarshal the config.Config from a confmap.Conf.
```
Comment (codeboten) [inline service/unmarshaler.go:35]:
```suggestion
// unmarshal the configSettings from a confmap.Conf.
```

File: service/internal/configunmarshaler/common.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// Copyright The OpenTelemetry Authors
```
Comment (codeboten) [inline service/internal/configunmarshaler/common.go:1]: I'd expect these in a file named `error.go` instead of common

**Review by codeboten — APPROVED** [review]
Just a couple of non-blocking comments

File: service/internal/configunmarshaler/common.go (hunk 2 above)
Comment (bogdandrutu) [reply service/internal/configunmarshaler/common.go:1]: Done.

File: service/unmarshaler.go (hunk 1 above)
Comment (bogdandrutu) [reply service/unmarshaler.go:35]: Done.

**Pull request merged**